### PR TITLE
Add offset

### PR DESCRIPTION
**Pull request by netanelC**
Adds offset feature

| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: src/common/providers/s3Provider.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ export class S3Provider implements Provider {
 
   private handleS3Error(filePath: string, error: unknown): never {
     let statusCode = httpStatus.INTERNAL_SERVER_ERROR;
-    let message = "Didn't throw a S3 exception in file";
+    let message = `Unexpected Error, Stringified: ${String(error)}`;
```
Comment (ronnahom96) [inline src/common/providers/s3Provider.ts]: You should check if it from type error with instanceOf and then use her message proprety to construct the message.

File: src/fileSyncerManager/fileSyncerManager.ts [hunk 2]
```diff
@@ -55,16 +55,19 @@ export class FileSyncerManager {
 
   private async sendFilesToCloudProvider(task: ITaskResponse<TaskParameters>): Promise<void> {
     this.logger.info({ msg: 'Starting sendFilesToCloudProvider' });
-    const filePaths: string[] = task.parameters.paths;
+    let counter = 0;
```
Comment (ronnahom96) [inline src/fileSyncerManager/fileSyncerManager.ts]: Consider renaming it to 'offsetNumber' or just 'offset' for a more appropriate name.

File: src/fileSyncerManager/fileSyncerManager.ts [hunk 3]
```diff
@@ -81,6 +84,20 @@ export class FileSyncerManager {
     }
   }
 
+  private async updateOffset(task: ITaskResponse<TaskParameters>, counter: number): Promise<void> {
+    const newOffset = task.parameters.offset + counter;
+    const payload: IUpdateTaskBody<TaskParameters> = {
+      status: OperationStatus.IN_PROGRESS,
+      parameters: { ...task.parameters, offset: newOffset }
+    };
+    try {
```
Comment (ronnahom96) [inline src/fileSyncerManager/fileSyncerManager.ts]: remove unimportant try & catch

**Review by ronnahom96 — CHANGES_REQUESTED** [review]

File: src/fileSyncerManager/fileSyncerManager.ts (hunk 2 above)
Comment (netanelC) [reply src/fileSyncerManager/fileSyncerManager.ts]: I think "offset" is just fine. There is no added value in "number".

File: src/fileSyncerManager/fileSyncerManager.ts (hunk 3 above)
Comment (netanelC) [reply src/fileSyncerManager/fileSyncerManager.ts]: Outdated

**Review by ronnahom96 — APPROVED** [review]

**Pull request merged**